### PR TITLE
Feat/server improv 1

### PR DESCRIPTION
**Pull request by MatheusCervieri**
I did some basic improovements to the server. Still need to do some things, but I already want to save it in the main.
Here is the list of the things i have done:

- Organized the code a little bit. Change the order of some functions, remove coments, change variable names, to be easy to read and understand. 
- Close all the fds that was open.
- Check for memory leaks - NO LEAKS ARE POSSIBLE.
- I change the read and write error handling to feet this: - Search for all read/recv/write/send and check if the returned value is correctly checked (checking only -1 or 0 values is not enough, both should be checked). 
- I was checking errno, and pdf explicit says that is forbiden. So i change it. 
- Put the Server Class in canonical form.
- Create a basic explanation of http server and I/O multiplexing and put it in docs. 
- I create a map container to store the readed request buffer. 
- I added some simple sh tests. 

Besides that, i make a important change that i think may help the tests. I change the timeout of the poll function:
```
int ret = poll( poll_fds, n_fds, 5000);
```
The poll function will wait 5000 miliseconds for a client request, if it dont receive any, it will close the server.

### CODE REVIEW

**Comment by vcwild** [comment]
@MatheusCervieri I need you to list what you are improving

File: sources/server/README.md [hunk 1]
```diff
@@ -1,3 +1,46 @@
+### Como um servidor http funciona?
```
Comment (vcwild) [inline sources/server/README.md]: Please move this README to `/docs`

File: sources/server/Server.cpp [hunk 2]
```diff
@@ -6,7 +6,7 @@
 /*   By: mvieira- <mvieira-@student.42sp.org.br>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/01/03 10:16:53 by mvieira-          #+#    #+#             */
```
Comment (vcwild) [inline sources/server/Server.cpp:8]: Delete 42 headers, as they clutter the git diffs with updates

Comment (iaurg) [reply sources/server/Server.cpp]: Removed all headers

**Review by vcwild — APPROVED** [review]
:medal_sports:

**Pull request merged**